airflow/operators/s3_to_hive_operator.py

```diff
@@ -146,16 +150,42 @@ def execute(self, context):
                 raise AirflowException(
                     "The key {0} does not exists".format(self.s3_key))
             s3_key_object = self.s3.get_key(self.s3_key)
+
+        if self.select_expression is not None and self.input_compressed\
```

Comment:
A couple of comments: 
- This is more of a nit, but I am more of a fan of using parentheses for line continuation rather than the backslash.
- `self.select_expression is not None` is used twice, time to maybe add a local var `has_select_expression = self.select_expression is not None`, or just use `if self.select_expression`, as none of the false-y values are valid select_expressions anyway. 
- Similarly move the test below `root, file_ext = os.path.splitext(s3_key_object.key)` and get the file extension parsed for free :)